pkg/controllers/depends_on.go

```diff
@@ -0,0 +1,26 @@
+package controllers
+
+import (
+	jobset "sigs.k8s.io/jobset/api/jobset/v1alpha2"
+)
+
+// dependencyReachedStatus checks if dependant ReplicatedJob reaches Ready or Complete status.
+func dependencyReachedStatus(dependsOnJob jobset.DependsOn, dependsOnJobReplicas int32, rJobsStatuses []jobset.ReplicatedJobStatus) bool {
+	// If the actual status is empty, return false.
+	actualStatus := findReplicatedJobStatus(rJobsStatuses, dependsOnJob.Name)
+	if actualStatus == nil {
+		return false
+	}
+
+	// For Complete status, number of replicas must be equal to number of succeeded Jobs.
+	if dependsOnJob.Status == jobset.DependencyComplete && dependsOnJobReplicas == actualStatus.Succeeded {
+		return true
+	}
+
```

Comment:
```suggestion
	if dependsOnJob.Status == jobset.DependencyComplete && dependsOnJobReplicas == actualStatus.Failed {
		return false
}
```

If we hit complete but the replicated job fail, what should we do?

Reply:
We won't hit complete here if ReplicatedJob fails.
Complete here means that [`.replicatedJobStatus.succeeded`](https://github.com/kubernetes-sigs/jobset/blob/c0bf2320b45a0ece4a480e9438760763e813f9b2/api/jobset/v1alpha2/jobset_types.go#L186) ==  [`.replicatedJob.replicas`](https://github.com/kubernetes-sigs/jobset/blob/c0bf2320b45a0ece4a480e9438760763e813f9b2/api/jobset/v1alpha2/jobset_types.go#L235)
Does it make sense @kannon92 ?